### DISCUSSION TITLE
Missing generator

### DISCUSSION THREAD

**Discussion by Noxillio** [airsquared/blobsaver#246] [Q&A] [answered]
I'm trying to use FutureRestore to go from 13.5.1 to 14.3.
I have saved every single blob for my phone from when I got it until now.
I have always specified the ApNonce, even before my device was jailbroken. I had it read from the device.

Am I just screwed out of using FutureRestore seeing as I can't get to the generator from any of the blobs I've saved using BlobSaver? Have I been doing several months of this for absolutely nothing? Or is the generator some kind of default? I don't know how the whole generator thing works but as far as I can tell BlobSaver uses 0x1111111111111111? Please help me.

**Comment by airsquared** [comment, marked as answer]
There is no generator used when you save blobs with a specific apnonce. The generator you will need to set is the same generator you had set on your device when you initially read the apnonce from your device.

**Reply by Noxillio** [reply]
I didn't have any generator set. I was unjailbroken. What do I do?

**Reply by airsquared** [reply]
If you didn't have any generator set because you were not jailbroken, then you have to try to restore without setting any generator.

Since the device was jailbroken at some point after reading the apnonce in an unjailbroken state, it changed the generator and the blobs saved with the apnonce read from the unjailbroken state will not be usable. If the device had never been jailbroken after initially reading the apnonce, the blobs still might have been usable.

**Reply by r1sk01** [reply]
@Noxillio I recommend restoring your rootfs, jailbreaking with something like Unc0ver or Odyssey and setting your generator in the jailbreak to the one in your blobs (I know this is necroposting but I want to help out)
